File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> players = new HashMap<>();
        
        for(String p : participant)
            players.put(p, players.getOrDefault(p, 0) + 1);
        
        for(String c : completion) {
            //players.put(c, players.getOrDefault(c, 0) - 1);
            int n = players.get(c) - 1;
            if(n == 0) players.remove(c);
            else players.put(c, n);
        }
        
        return players.keySet().iterator().next();
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <bits/stdc++.h>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    // 동명이인 처리가 핵심임.
    sort(participant.begin(), participant.end());
    sort(completion.begin(), completion.end());
    for(int i = 0 ; i < completion.size(); i++){
        if(completion[i] != participant[i]){
            return participant[i];
        }
    }
    return participant[participant.size() - 1];
}